### PR TITLE
Add tests for components

### PR DESCRIPTION
**Pull request by carpenterd777**
Adds some more unit tests for each component. Most of the tests are focused on making sure each component renders the text that we expect them to. Next, I'll add tests for the custom hooks we've made so far.

Also adds codecov support. Once we know this is working, I'll add a badge in the README.

### CODE REVIEW

File: src/components/NearbyPlaces/NearbyPlaces.tsx [hunk 1]
```diff
@@ -27,6 +27,7 @@ export const NearbyPlaces: React.FC<{ stopStrolling: () => void }> = ({ stopStro
 								<Text
 									key={place.name}
 									style={{ fontSize: 24, fontWeight: "bold", marginBottom: 15 }}
+									testID={`nearby-place-${index}`}
```
Comment (aej11a) [inline src/components/NearbyPlaces/NearbyPlaces.tsx:30]: Have you seen this? https://reactnative.dev/docs/testing-overview

> As a rule of thumb, prefer using things users can see or hear: make assertions using rendered text or accessibility helpers. Conversely, you should avoid: making assertions on component props or state, testID queries

File: tests/NearbyPlaces.test.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+		photo_reference: "a",
+		height: 100,
+		width: 100,
+		html_attributions: [],
+	},
+];
+const mockPlaces: Partial<PlaceData & { rating: number }>[] = [
+	{
+		name: "The Absolute Best Place in the Whole Wide World",
+		formatted_address: "312 Jones Pl",
```
Comment (aej11a) [inline tests/NearbyPlaces.test.tsx:52]: damn, how did you know the name of my place???

File: src/components/NearbyPlaces/NearbyPlaces.tsx (hunk 1 above)
Comment (carpenterd777) [reply src/components/NearbyPlaces/NearbyPlaces.tsx:30]: yeah, now that I think about it there is a better way to do this

**Comment by aej11a** [comment]
Looks much better now. I'm still a little cynical about the value of feeding in data and being like "this test expects the data we just fed in to render" but your mocks and everything seem high quality so merging

**Review by aej11a — APPROVED** [review]

**Pull request merged**